rng/src/main.rs

```diff
@@ -0,0 +1,241 @@
+
+        let handle = thread::spawn(move || loop {
+            // If creating the VuRngBackend isn't successull there isn't much else to do than
+            // killing the thread, which .unwrap() does.  When that happens an error code is
+            // generated and displayed by the runtime mechanic.  Killing a thread doesn't affect
+            // the other threads spun-off by the daemon.
+            let vu_rng_backend = Arc::new(RwLock::new(
+                VuRngBackend::new(random.clone(), period_ms, max_bytes).unwrap(),
+            ));
+
+            let mut daemon = VhostUserDaemon::new(
+                String::from("vhost-user-RNG-daemon"),
+                vu_rng_backend.clone(),
+                GuestMemoryAtomic::new(GuestMemoryMmap::new()),
+            )
+            .unwrap();
+
+            let listener = Listener::new(socket.clone(), true).unwrap();
+            daemon.start(listener).unwrap();
+
+            match daemon.wait() {
+                Ok(()) => {
+                    info!("Stopping cleanly.");
+                }
+                Err(vhost_user_backend::Error::HandleRequest(
+                    vhost_user::Error::PartialMessage,
+                )) => {
+                    info!("vhost-user connection closed with partial message. If the VM is shutting down, this is expected behavior; otherwise, it might be a bug.");
+                }
+                Err(e) => {
+                    warn!("Error running daemon: {:?}", e);
+                }
+            }
+
+            // No matter the result, we need to shut down the worker thread.
+            vu_rng_backend
+                .read()
+                .unwrap()
+                .exit_event
+                .write(1)
+                .expect("Shutting down worker thread");
+        });
+
+        handles.push(handle);
+    }
+
+    for handle in handles {
+        handle.join().map_err(|_| Error::FailedJoiningThreads)?;
+    }
+
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+
+    start_backend(VuRngConfig::try_from(RngArgs::parse()).unwrap())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+
+    #[test]
+    fn verify_cmd_line_arguments() {
+        // All parameters have default values, except for the socket path.
+        let default_args: RngArgs = Parser::parse_from(&["", "-s/some/socket_path"]);
+
+        // A valid configuration that should be equal to the above default configuration.
+        let args = RngArgs {
+            period: VHU_RNG_MAX_PERIOD_MS,
+            max_bytes: usize::MAX,
+            socket_count: 1,
+            socket_path: "/some/socket_path".to_string(),
+            rng_source: "/dev/urandom".to_string(),
+        };
+
+        // All configuration elements should be what we expect them to be.
+        assert_eq!(default_args, args);
+
+        // All elements are valid.
+        assert!(VuRngConfig::try_from(default_args).is_ok());
```

Comment:
Here .unwrap!() returns a struct VuRngConfig, which won't work with assert!() because the latter requires a boolean argument.  Using .unwrap!() with assert_eq!() would require to build a VuRngConfig to compare with, something that is a little overboard when we only want to test for a valid condition.

Reply:
I ended up having to use assert_eq!() with VuRngConfig::try_from() to exercise the new .trim() functionality, making the original assert!() test useless.